pkg/segment/search/searchaggs.go

```diff
@@ -656,20 +656,25 @@ func PerformMeasureAggsOnRecs(nodeResult *structs.NodeResult, recs map[string]ma
 
 		for colName, value := range searchResults.GetSegmentStatsMeasureResults() {
 			finalCols[colName] = true
-			if value.Dtype == utils.SS_DT_FLOAT {
+			switch value.Dtype {
+			case utils.SS_DT_FLOAT:
 				value.CVal = humanize.CommafWithDigits(value.CVal.(float64), 3)
-			} else if value.Dtype == utils.SS_DT_STRING_SLICE {
+			case utils.SS_DT_STRING_SLICE:
 				strVal, err := value.GetString()
 				if err != nil {
 					log.Errorf("PerformMeasureAggsOnRecs: failed to obtain string representation of slice %v: %v", value, err)
+					value.CVal = ""
+				} else {
+					value.CVal = strVal
 				}
-				value.CVal = strVal
-			} else {
+			case utils.SS_DT_SIGNED_NUM:
 				value.CVal = humanize.Comma(value.CVal.(int64))
+			default:
+				log.Errorf("PerformMeasureAggsOnRecs: Unexpected type %v ", value.Dtype)
+				value.CVal = ""
 			}
```

Comment:
Thanks for adding this extra failsafe. However, setting the `CVal` and not setting the `Dtype` could potentially cause problems (e.g., if we got to this `default` because the `Dtype` is `SS_DT_BOOL`, but now `CVal` has a string, so later we might try to extract the value as a boolean and panic when it's actually a string). I think a few good options are:
1. Return an error here
2. Set the `Dtype` to `SS_INVALID` (then the `CVal` value doesn't matter)

Reply:
Updated the code. Dtype is set to SS_INVALID  and is ignored in case of errors.